rlclientlib/logger/async_batcher.h

```diff
@@ -105,6 +109,7 @@ namespace reinforcement_learning { namespace logger {
       }
     }
     
+    if (_events_counter_status == events_counter_status::ENABLE) { evt.set_event_index(_event_index); }
```

Comment:
actually there is a problem here:
increment_counter_from_thread_1 -> increment_counter_from_thread_2 -> set_event_index_from_thread1 -> set_event_index_from_thread_2
in this case both events will get the same index

Reply:
now incrementing the event_index and updating it in event are performed under lock so no two threads can access simultaneously